### ISSUE TITLE
dart and random forest cannot correctly use MAPE objective

### ISSUE THREAD

**Issue by zhongxurpac** [Microsoft/LightGBM#1579]
I'm trying to use random forest and dart to make regression prediction.
Due to competition requirement, I shifted my labels by 2, range from 0-1 shifted to 2-3.
(like the original label is 0.4, now it's 2.4)
However, when I use random forest/dart as boosting, MAE/mape as objective to make prediction, the prediction still falls in the range of 0-1.
I tried random forest as boosting, rmse as objective. The result successfully fell into the range of 2-3.
I tried gbdt as boosting. The result also successfully fell into the range of 2-3.

Below is my parameter, I change rf to gbdt for test.

params = {
      'task'                 : 'train',
      'boosting_type'        : 'rf',
      'objective'            : 'mape',
      'learning_rate'        : 0.01,
      'metric'               : {'mape'},
      'n_estimators'         : 500,
      'num_leaves'           : 5000,
      'max_depth'            : 10,
      'feature_fraction'     : 0.8,
      'feature_fraction_seed': 2,
      'bagging_fraction'     : 0.8,
      'min_data_in_leaf'     : 1,
      'verbose'              : 0,
      'bagging_freq'         : 1,
      'early_stopping_round': 500
    }
C++/Python/R version:
python 36

I think there might be something not going well between tree models and MAE/MAPE objective?
Help appreciated!

**Comment by LucasMaron** [comment]
I realized this issue. They are currently incompatible. 
I'm looking into this and will be back with some test results soon.

**Comment by StrikerRUS** [comment]
ping @guolinke
